packages/components/src/components/tooltip/Tooltip.tsx

```diff
@@ -65,140 +53,123 @@ const Tooltip: React.FC<TooltipProps> = ({
   placement,
   type,
   visible,
-  hoverTimeout,
   className,
   wrapperClassName,
   hoverDelay = 0,
+  hoverTimeout = 100,
   ...otherProps
 }) => {
-  const [popperElement, setPopperElement] = useState(null);
-  const [referenceElement, setReferenceElement] = useState(null);
+  const arrowRef = React.useRef(null);
+  const [isOpen, setIsOpen] = useState(visible);
 
-  const [showHover, setShowHover] = useState(false);
-  const [showClick, setShowClick] = useState(false);
-  const handleMouseMove = useMemo(() => debouncer(setShowHover, hoverDelay, hoverTimeout), [hoverDelay, hoverTimeout]);
-
-  const ref = useOnclickOutsideCool(() =>
-  { setShowClick(false) },
-  { disabled: !showClick });
-
-  const { styles, attributes } = usePopper(referenceElement, popperElement, {
-    placement: placement || 'top',
-    modifiers: [
-      {
-        name: 'flip',
-        options: {
-          fallbackPlacements: ['bottom', 'right', 'left'],
-        },
-      },
-      {
-        name: 'offset',
-        options: {
-          offset: [0, type === 'hint' ? 15 : 9],
-        },
-      },
+  React.useEffect(() => {
+    setIsOpen(visible)
+  }, [visible])
+  
+  const { refs, floatingStyles, context } = useFloating({
+    middleware: [
+      arrow({
+        element: arrowRef,
+      }),
+      flip(),
+      offset(type === 'hint' ? 15 : 9),
     ],
+    open: isOpen,
+    onOpenChange: setIsOpen,
+    placement: placement ?? 'top',
+    whileElementsMounted: autoUpdate,
   });
 
-  const isVisible =
-    typeof visible !== 'undefined'
-      ? visible
-      : displayTrigger === 'hover'
-        ? showHover
-        : showClick;
+  const dismiss = useDismiss(context, {
+    enabled: displayTrigger === 'click',
+  });
+  
+  const hover = useHover(context, {
+    delay: {
+      close: hoverTimeout,
+      open: hoverDelay,
+    },
+    enabled: displayTrigger === 'hover'
+  });
 
-  const children = <div
-    className={ clsx('tk-tooltip__wrapper', wrapperClassName) }
-    ref={ composeRefs(setReferenceElement, ref) }
-  >
-    {otherProps.children}
-  </div>;
+  const { getReferenceProps, getFloatingProps } = useInteractions([
+    dismiss,
+    hover,
+  ])
 
-  return (
-    <>
-      { /**
-       * The element tooltip is wrapped around.
-       */ }
-      { /** Extra span to make sure Tooltip works on disabled children */ }
-      {displayTrigger === 'hover' &&
-          showTooltipOnHover(children, handleMouseMove)}
-      {displayTrigger === 'click' &&
-          showTooltipOnClick(children, showClick, setShowClick)}
```

Comment:
Before clicking anywhere on a hint would close it if displayTrigger was `click` which is incorrect behaviour. Now it'll only close if close label is clicked, the reference element or if you click outside.